File: lecture notes/matlab/lec4/run1.m
clear;clc;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
pvec=1:1:5;
tsvec=zeros(size(pvec));

for i=1:length(pvec)
    pval=pvec(i);
    Gs=tf(pval,[1 pval]);
    info=stepinfo(Gs);
    tsvec(i)=info.SettlingTime;

end

figure(1);clf;hold on;grid minor;
set(gca,'MinorGridColor','k','MinorGridAlpha',1);
xlabel("p");ylabel("t_s");legend("show");
plot(pvec,tsvec,'k','LineWidth',2,'DisplayName','stepinfo');
plot(pvec,4./pvec,'b','LineWidth',2,'DisplayName','4/p');
print("../../img/"+"lec4_plot1.eps",'-depsc','-r150');

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
wn=2;
zetavec=0.1:0.1:1;

tsvec=zeros(size(zetavec));

for i=1:length(zetavec)
    zetaval=zetavec(i);
    Gs=tf(wn^2,[1 2*zetaval*wn wn^2]);
    info=stepinfo(Gs);
    tsvec(i)=info.SettlingTime;
end

figure(2);clf;hold on;grid minor;
set(gca,'MinorGridColor','k','MinorGridAlpha',1);
xlabel("\zeta w_n");ylabel("t_s");legend("show");

plot(zetavec*wn,tsvec,'k','LineWidth',2,'DisplayName','stepinfo');
plot(zetavec*wn,4./(zetavec*wn),'b','LineWidth',2,'DisplayName','4/a');
print("../../img/"+"lec4_plot2.eps",'-depsc','-r150');

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
wnvec=1:5;
zeta=0.6;

tsvec=zeros(size(wnvec));

for i=1:length(wnvec)
    wnval=wnvec(i);
    Gs=tf(wn^2,[1 2*zeta*wnval wnval^2]);
    info=stepinfo(Gs);
    tsvec(i)=info.SettlingTime;
end

figure(3);clf;hold on;grid minor;
set(gca,'MinorGridColor','k','MinorGridAlpha',1);
xlabel("\zeta w_n");ylabel("t_s");legend("show");
plot(wnvec*zeta,tsvec,'k','LineWidth',2,'DisplayName','stepinfo');
plot(wnvec*zeta,4./(zeta*wnvec),'b','LineWidth',2,'DisplayName','4/a');

print("../../img/"+"lec4_plot3.eps",'-depsc','-r150');

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
wn=2;
zetavec=0.1:0.1:1;

osvec=zeros(size(zetavec));

for i=1:length(zetavec)
    zetaval=zetavec(i);
    Gs=tf(wn^2,[1 2*zetaval*wn wn^2]);
    info=stepinfo(Gs);
    osvec(i)=info.Overshoot;
end

figure(2);clf;hold on;grid minor;
set(gca,'MinorGridColor','k','MinorGridAlpha',1);
xlabel("\zeta");ylabel("asim");legend("show");
plot(zetavec,osvec,'k','LineWidth',2,'DisplayName','stepinfo');
plot(zetavec,100*exp(-pi*zetavec./sqrt(1-zetavec.^2)),'b','LineWidth',2,'DisplayName','100e^{-pi\\zeta/\\sqrt{1-\\zeta^2}}');

print("../../img/"+"lec4_plot4.eps",'-depsc','-r150');

File: lecture notes/python/lec4/run1.py
import matplotlib.pyplot as plt
import numpy as np
import sympy as sym
import tbcontrol
import control
############################
# q1
############################
def fun_q1():
    pvec=np.arange(1,5+1,1)
    tsvec=np.zeros(pvec.shape)
    for i in range(0,len(pvec)):
        pval=pvec[i]
        Gs=control.tf(pval,np.array([1,pval]))
        info=control.step_info(Gs)
        tsvec[i]=info['SettlingTime']

    plt.grid('minor')
    plt.xlabel("p")
    plt.ylabel("ts")
    plt.title("p ile ts arasındaki ilişki")

    plt.plot(pvec,tsvec,'b')
    plt.show()
############################
# q2
############################
def fun_q2():
    wn=2
    zetavec=np.arange(0.1,1,0.1)
    tsvec=np.zeros(zetavec.shape)
    tsformula=np.zeros(zetavec.shape)
    for i in range(0,len(zetavec)):
        zetaval=zetavec[i]
        Gs=control.tf(wn**2,np.array([1,2*zetaval*wn,wn**2]))
        info=control.step_info(Gs)
        tsvec[i]=info['SettlingTime']
        tsformula[i]=4/(zetaval*wn)

    plt.grid('minor')
    plt.xlabel("zeta wn")
    plt.ylabel("ts")
    plt.title("zeta wn ile ts arasındaki ilişki")


    plt.plot(zetavec*wn,tsformula,'k')
    plt.plot(zetavec*wn,tsvec,'b')
    plt.show()
############################
# q3
############################
def fun_q3():
    zeta=0.6
    wnvec=np.arange(1,5+1,1)
    tsvec=np.zeros(wnvec.shape)
    tsformula=np.zeros(wnvec.shape)
    for i in range(0,len(wnvec)):
        wnval=wnvec[i]
        Gs=control.tf(wnval**2,np.array([1,2*zeta*wnval,wnval**2]))
        info=control.step_info(Gs)
        tsvec[i]=info['SettlingTime']
        tsformula[i]=4/(zeta*wnval)

    plt.grid('minor')
    plt.xlabel("zeta wn")
    plt.ylabel("ts")
    plt.title("zeta wn ile ts arasındaki ilişki")


    plt.plot(zeta*wnvec,tsformula,'k')
    plt.plot(zeta*wnvec,tsvec,'b')
    plt.show()
############################
# q4
############################
def fun_q4():
    wn=2
    zetavec=np.arange(0.1,1,0.1)
    osvec=np.zeros(zetavec.shape)
    osformula=np.zeros(zetavec.shape)
    for i in range(0,len(zetavec)):
        zetaval=zetavec[i]
        Gs=control.tf(wn**2,np.array([1,2*zetaval*wn,wn**2]))
        info=control.step_info(Gs)
        osvec[i]=info['Overshoot']
        osformula[i]=100*np.exp(-np.pi*zetaval/np.sqrt(1-zetaval**2))

    plt.grid('minor')
    plt.xlabel("zeta")
    plt.ylabel("ts")
    plt.title("zeta ile aşım arasındaki ilişki")


    plt.plot(zetavec*wn,osformula,'k',linewidth=3)
    plt.plot(zetavec*wn,osvec,'b')
    plt.show()
fun_q4()